draft-ietf-masque-h3-datagram.md

```diff
@@ -570,6 +570,20 @@ all cases, the maximum permitted value of the H3_DATAGRAM SETTINGS parameter is
 1.
 
 
+## Note About Draft Versions
+
+\[\[RFC editor: please remove this section before publication.]]
+
+Some revisions of this draft specification use a different value for the
+H3_DATAGRAM Settings Parameter. This allows new draft revisions to make
+incompatible changes. Implementations MAY support multiple draft versions at
+the same time. In that scenario, endpoints MUST send multiple values for
+H3_DATAGRAM. Once an endpoint has sent and received SETTINGS, it MUST compute
+the intersection of the values it has sent and received, and then it MUST
+select and use the most recent draft version from the intersection set. This
```

Comment:
Do we want to say anything about when the intersection is empty? Perhaps it's useful to close the connection with a specific error code like "H3_DATAGRAM_REQUIRED" or something. The receiving endpoint can then use that to see if it is due to mismatched sets, or that the peer was hoping for datagrams where they'll never be supported.

Reply:
I think that degrades easily to the case where the peer doesn't support datagram? Adding that error sounds reasonable, but is orthogonal to the multiple drafts question. Maybe file an issue and we can bring it up at 111?

Reply:
I'm happy to punt a special close code to a new issue but I think we need to fix this text up a bit more. The more i think about it, the more I'm unsure what the intent is.

Section 6 doesn't do a brilliant job of stating if both ends have to send a setting with value 1 in order for H3 Datagram to function or not. Theoretically, you could use true QUIC datagrams in one direction and datagram capsules in another (I'm not vouching to actually do that fwiw).

Extending to this PR, all one way to interpret this is that only the versions have to intersect. Not the values (0 or 1). It's implied that an endpoint advertises support with 1. But if I send you "old 0" and "new 1" and you send "old 1" then interop might fail because we agreed on a version but not how to use it.

Reply:
Discussed offline, let's merge this and have a follow up to clean up the exact semantics of the setting